api/src/main/java/edu/cornell/mannlib/vitro/webapp/servlet/setup/ConfigurationModelsSetup.java

```diff
@@ -31,35 +45,166 @@ public void contextInitialized(ServletContextEvent sce) {
 			setupModel(ctx, DISPLAY_TBOX, "displayTbox");
 			setupModel(ctx, DISPLAY_DISPLAY, "displayDisplay");
 			setupModel(ctx, USER_ACCOUNTS, "auth");
-			ss.info(this,
-					"Set up the display models and the user accounts model.");
+			ss.info(this, "Set up the display models and the user accounts model.");
 		} catch (Exception e) {
 			ss.fatal(this, e.getMessage(), e.getCause());
 		}
 	}
 
-	private void setupModel(ServletContext ctx, String modelUri,
-			String modelPath) {
+	private void setupModel(ServletContext ctx, String modelUri, String modelPath) {
 		try {
 			OntModel ontModel = ModelAccess.on(ctx).getOntModel(modelUri);
-			loadFirstTimeFiles(ctx, modelPath, ontModel);
+			if (ontModel.isEmpty()) {
+				loadFirstTimeFiles(ctx, modelPath, ontModel);
+				// backup firsttime files
+				OntModel baseModelFirsttime = ModelAccess.on(ctx).getOntModel(modelUri + "Firsttime");
+				baseModelFirsttime.add(ontModel);
+			} else {
+				// Check if the firsttime files have changed since the firsttime startup,
+				// if so, then apply the changes but not overwrite the whole user model
+				applyFirstTimeChanges(ctx, modelPath, modelUri, ontModel);
+			}
+
 			loadEveryTimeFiles(ctx, modelPath, ontModel);
 		} catch (Exception e) {
-			throw new RuntimeException("Failed to create the '" + modelPath
-					+ "' model (" + modelUri + ").", e);
+			throw new RuntimeException("Failed to create the '" + modelPath + "' model (" + modelUri + ").", e);
 		}
 	}
 
-	private void loadFirstTimeFiles(ServletContext ctx, String modelPath,
-			OntModel baseModel) {
+	private void loadFirstTimeFiles(ServletContext ctx, String modelPath, OntModel baseModel) {
 		RDFFilesLoader.loadFirstTimeFiles(ctx, modelPath, baseModel, baseModel.isEmpty());
 	}
 
-	private void loadEveryTimeFiles(ServletContext ctx, String modelPath,
-			OntModel memoryModel) {
+	private void loadEveryTimeFiles(ServletContext ctx, String modelPath, OntModel memoryModel) {
 		RDFFilesLoader.loadEveryTimeFiles(ctx, modelPath, memoryModel);
 	}
 
+	/*
+	 * Check if the firsttime files have changed since the firsttime startup, if so,
+	 * then apply the changes but not overwrite the whole user model
+	 */
+	private void applyFirstTimeChanges(ServletContext ctx, String modelPath, String modelUri, OntModel userModel) {
+
+		log.info("Start with check: Reload firsttime files on start-up if changed");
```

Comment:
Could you please change the logging message as follows:
```
log.info("Reload firsttime files on start-up if changed: '" + modelPath +"', URI: '" +modelUri+ "'");
```

Reply:
i changed it